apps/e2e/cypress/e2e/statusActions.cy.ts

```diff
@@ -0,0 +1,228 @@
+    cy.get('[data-cy="PI-email-template"] input')
+      .invoke('val')
+      .should('not.be.empty');
+
+    cy.get('[data-cy="action-recipient-CO_PROPOSERS"] input').click();
+
+    cy.get('[data-cy="submit"]').contains('Add status actions').click();
+
+    cy.get('[data-cy="CO_PROPOSERS-email-template"] input').should(
+      'be.focused'
+    );
+    cy.get<JQuery<HTMLInputElement>>(
+      '[data-cy="CO_PROPOSERS-email-template"] input'
+    ).then(($input) => {
+      expect($input[0].validity.valid).to.be.false;
+      expect($input[0].validationMessage).to.include(
+        'Please fill out this field.'
+      );
+    });
+
+    cy.get('[data-cy="CO_PROPOSERS-email-template"] input').click();
+    cy.get('.MuiAutocomplete-listbox li').last().click();
+
+    cy.get('[data-cy="submit"]').contains('Add status actions').click();
+
+    cy.notification({ variant: 'success', text: 'success' });
+  });
+
+  it('User Officer should be able to delete a status action added to the workflow connection', () => {
+    const statusActionConfig = {
+      recipientsWithEmailTemplate: [
+        {
+          recipient: { name: EmailStatusActionRecipients.PI, description: '' },
+          emailTemplate: { id: 'pi-template', name: 'PI template' },
+        },
+      ],
+    };
+
+    cy.addProposalWorkflowStatus({
+      droppableGroupId: initialDBData.workflows.defaultDroppableGroup,
+      proposalStatusId: initialDBData.proposalStatuses.feasibilityReview.id,
+      proposalWorkflowId: initialDBData.workflows.defaultWorkflow.id,
+      sortOrder: 1,
+      prevProposalStatusId: initialDBData.proposalStatuses.draft.id,
+    }).then((result) => {
+      cy.addConnectionStatusActions({
+        actions: [
+          {
+            actionId: 1,
+            actionType: ProposalStatusActionType.EMAIL,
+            config: JSON.stringify(statusActionConfig),
+          },
+        ],
+        connectionId: result.addProposalWorkflowStatus.id,
+        workflowId: initialDBData.workflows.defaultWorkflow.id,
+      });
+    });
+
+    cy.login('officer');
+    cy.visit('/ProposalWorkflowEditor/1');
+
+    cy.finishedLoading();
+
+    cy.get(`[data-cy^="connection_FEASIBILITY_REVIEW"]`).click();
+
+    cy.get('[data-cy="status-events-and-actions-modal"]')
+      .contains('Status actions')
+      .click();
+
+    cy.finishedLoading();
+
+    cy.get('[data-cy="EMAIL-status-action"] input').should('be.checked');
+
+    cy.get('[data-cy="EMAIL-status-action"] input').uncheck();
+
+    cy.get('[data-cy="submit"]').contains('Add status actions').click();
+
+    cy.notification({ variant: 'success', text: 'success' });
+
+    cy.closeModal();
+
+    cy.get(
+      `[data-cy^="connection_FEASIBILITY_REVIEW"] [data-testid="PendingActionsIcon"]`
+    ).should('not.exist');
+  });
+  // TODO: Once the testing environment for email sending is in place we can extend the test cases with testing the actual run of the status actions.
```

Comment:
May I ask how you propose testing if the action is invoked?

Reply:
Maybe we should introduce event logs in the workflows and write that to the database when email action is invoked and it was run successfully. Need to think little bit better about this.